### ISSUE TITLE
in the arm windows10,can't create virtual network adapter

### ISSUE THREAD

**Issue by reheroad** [SoftEtherVPN/SoftEtherVPN#1331]
I am using version 4.34, my operating system is macos (m1 chip), I use Parallels Desktop to create an arm windows10, I cannot create a virtual network adapter。
in the client_log,don't find error logs.
If you can get help, I am very grateful.

**Comment by davidebeatrici** [comment]
We currently don't provide drivers for ARM64, however I will work on it soon.

**Comment by ronylee** [comment]
Do you have a driver for arm64 now ？

**Comment by davidebeatrici** [comment]
Unfortunately not, however I now have an ARM64 machine I can test on.

**Comment by hljinjiang** [comment]
Hope to support MAC M1 soon

**Comment by AshWhitear** [comment]
any updates on this one? I'd happily give an early version a crack, or even a use workaround

**Comment by JanefonYang** [comment]
Do you have a driver for arm64 now ？

**Comment by JanefonYang** [comment]
Do you have a driver for arm64 now ？

**Comment by muuvmuuv** [comment]
Is anything needed for supporting ARM based chips?

**Comment by chipitsine** [comment]
> Is anything needed for supporting ARM based chips?

we need a developer with ARM hardware to test himself.

**Comment by muuvmuuv** [comment]
I have a newly bought M3 Pro, I could test it against an IPSec IKEv1 VPN. Where do I start?

**Comment by chipitsine** [comment]
> I have a newly bought M3 Pro, I could test it against an IPSec IKEv1 VPN. Where do I start?

it would be nice if you test and confirm whether it works or not

**Comment by muuvmuuv** [comment]
I will have a look at it next week. Have to setup the project first.

**Comment by muuvmuuv** [comment]
So, these instructions here https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md#build-from-source-code-and-install just work and doing an "./build/vpnclient start" produces no errors. Currently trying to connect to our VPN server.

**Comment by 19byte** [comment]
> So, these instructions here https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md#build-from-source-code-and-install just work and doing an "./build/vpnclient start" produces no errors. Currently trying to connect to our VPN server.

is it working?

**Comment by jacksonattwood** [comment]
I'll have a look and see what I can get to work:
Win11 24H2, Snapdragon arm64 if you're interested

**Comment by jacksonattwood** [comment]
Yeah so I couldn't get it to build, Visual Studio is a pain. If someone could give me some pointers, I'll put some extra effort in 👍

**Comment by talynone** [comment]
> > Yeah so I couldn't get it to build, Visual Studio is a pain. If someone could give me some pointers, I'll put some extra effort in 👍
> 
> I tried to build it on my surface pro 11, I remembered the only things need to change is the NDIS driver and after some simple modification it could work, but the only problem is we could not sign the driver with a valid certifacte, and the driver could only be loaded if you unlock the secure boot and the Windows should be in developer mode.
> 
> Support for Windows arm64 is better be done by someone that have the permission to sign the driver, otherwise the driver could not be used by normal users.

Could you submit your modifications as a pull request?

**Comment by xuzhengyi1995** [comment]
> > > Yeah so I couldn't get it to build, Visual Studio is a pain. If someone could give me some pointers, I'll put some extra effort in 👍
> > 
> > 
> > I tried to build it on my surface pro 11, I remembered the only things need to change is the NDIS driver and after some simple modification it could work, but the only problem is we could not sign the driver with a valid certifacte, and the driver could only be loaded if you unlock the secure boot and the Windows should be in developer mode.
> > Support for Windows arm64 is better be done by someone that have the permission to sign the driver, otherwise the driver could not be used by normal users.
> 
> Could you submit your modifications as a pull request?

Currently my change is too rough to create a PR, I will do some more work to ensure the code quality and try to submit a PR for it. Win 11 arm64 does not support NDIS 6.2 so it's better to create a new driver folder in this project, I will try to work on it.
